core-java-modules/core-java-io-5/src/test/java/com/baeldung/outputstreamtobytearray/OutputStreamToByteArrayUnitTest.java

```diff
@@ -0,0 +1,65 @@
+package com.baeldung.outputstreamtobytearray;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class OutputStreamToByteArrayUnitTest {
+
+    @Test
+    public void givenFileOutputStream_whenUsingFileUtilsToReadTheFile_thenReturnByteArray(@TempDir Path tempDir) throws IOException {
+        String data = "Welcome to Baeldung!";
+        String fileName = "file.txt";
+        Path filePath = tempDir.resolve(fileName);
+
+        try (FileOutputStream outputStream = new FileOutputStream(filePath.toFile())) {
+            outputStream.write(data.getBytes(StandardCharsets.UTF_8));
+        }
+
+        byte[] writtenData = FileUtils.readFileToByteArray(filePath.toFile());
+        String result = new String(writtenData, StandardCharsets.UTF_8);
+        assertEquals(data, result);
+    }
+
+
+    @Test
+    public void givenByteArrayOutputStream_whenUsingDrainableOutputStream_thenReturnByteArray() throws IOException {
+        String data = "Welcome to Baeldung!";
+
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        DrainableOutputStream drainableOutputStream = new DrainableOutputStream(byteArrayOutputStream);

```

Comment:
I think it could be confusing to use a ByteArrayOutputStream given that we also use that in the DrainableOutputStream implementation. How about we just use System.out instead as an example?
```suggestion
        DrainableOutputStream drainableOutputStream = new DrainableOutputStream(System.out);
```